src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/process-form.service.ts

```diff
@@ -2,22 +2,25 @@ import { Injectable } from "@angular/core";
 import { FormGroup } from "@angular/forms";
 import { SchemaType } from "../form-components/schema-field/schema-field.component";
 import { SchemaControlType, OrderControlType, SourceOrder } from "./process-form.service.types";
-import { SetPollingSource } from "src/app/api/kamu.graphql.interface";
+import { AddPushSource, SetPollingSource } from "src/app/api/kamu.graphql.interface";
 import { SetPollingSourceSection } from "src/app/shared/shared.types";
 import { AddPollingSourceEditFormType, FetchKind, PrepareKind } from "./add-polling-source-form.types";
 import AppValues from "src/app/common/app.values";
+import { has } from "lodash";
 
 @Injectable({
     providedIn: "root",
 })
 export class ProcessFormService {
     public transformForm(formGroup: FormGroup): void {
         this.transformSchema(formGroup);
-        this.processFetchOrderControl(formGroup);
+        if (has(formGroup.value, "fetch")) {
+            this.processFetchOrderControl(formGroup);
+            this.processEventTimeControl(formGroup);
+            this.processPipeCommandControl(formGroup);
+        }
```

Comment:
This correction is clearly to handle push sources, but the service itself is placed within `add-polling-source` folder.

Reply:
Moved to a separate folder